src/js/components/invoice/AddItemsPage.jsx

```diff
@@ -337,6 +361,53 @@ class AddItemsPage extends Component {
       });
   }
 
+  validate(values) {
+    const errors = {};
+    errors.invoiceItems = [];
+
+    _.forEach(values?.invoiceItems, (item, key) => {
+      if (_.has(item, 'isValid') && !item.isValid) {
+        errors.invoiceItems[key] = { quantity: item?.errorMessage };
+      }
+    });
+
+    return errors;
+  }
+
+  async validateInvoiceItem({
+    invoiceItem,
+    rowIndex,
+  }) {
+    this.debouncedInvoiceItemValidation.cancel();
+
+    await invoiceItemApi.validateInvoiceItem(invoiceItem)
+      .then(() => {
+        const updatedValues = update(this.state.values, {
```

Comment:
I honestly hate this `update` method and everything from this package, I'd prefer a simple `.map`, but saw the change request....